llvm/lib/Target/AIE/AIEBaseSubtarget.cpp

```diff
@@ -469,6 +494,20 @@ class WAWEdges : public ScheduleDAGMutation {
       }
       LiveRegs.stepBackward(MI);
     }
+    // Updating liveins of current region
```

Comment:
As said above, I believe this can be dangerous. I'd at least like to move that to another commit to keep the code changes as small as possible.